code/modules/crew_objectives/medical_objectives.dm

```diff
@@ -2,94 +2,101 @@
+		/area/medical/exam_room,
+		/area/medical/medbay/aft,
+		/area/medical/medbay/central,
+		/area/medical/medbay/lobby,
+		/area/medical/patients_rooms,
+		/area/medical/sleeper,
+		/area/medical/storage
+	))
 
 /datum/objective/crew/morgue/check_completion()
-	var/list/medical_areas = typecacheof(list(/area/medical/cryo, /area/medical/genetics/cloning, /area/medical/exam_room,
-		/area/medical/medbay/aft, /area/medical/medbay/central, /area/medical/medbay/lobby, /area/medical/patients_rooms,
-		/area/medical/sleeper, /area/medical/storage))
+	if(..())
+		return TRUE
 	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)
 		var/area/A = get_area(H)
 		if(H.stat == DEAD && is_station_level(H.z) && is_type_in_typecache(A, medical_areas)) // If person is dead and corpse is in one of these areas
-			return ..()
+			return FALSE
 	return TRUE
 
 /datum/objective/crew/paramedicmorgue
 	explanation_text = "Ensure that no corpses remain outside of Medbay when the shift ends."
-	jobs = "paramedic"
+	jobs = JOB_NAME_PARAMEDIC
+	var/static/list/medical_areas_morgue = typecacheof(list(
+		/area/medical/cryo,
+		/area/medical/genetics/cloning,
+		/area/medical/exam_room,
+		/area/medical/medbay/aft,
+		/area/medical/medbay/central,
+		/area/medical/medbay/lobby,
+		/area/medical/patients_rooms,
+		/area/medical/sleeper,
+		/area/medical/storage,
+		/area/medical/morgue
+	))
 
 /datum/objective/crew/paramedicmorgue/check_completion()
-	var/list/medical_areas_morgue = typecacheof(list(/area/medical/cryo, /area/medical/genetics/cloning, /area/medical/exam_room,
-		/area/medical/medbay/aft, /area/medical/medbay/central, /area/medical/medbay/lobby, /area/medical/patients_rooms,
-		/area/medical/sleeper, /area/medical/storage, /area/medical/morgue))
+	if(..())
+		return TRUE
 	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)
 		var/area/A = get_area(H)
 		if(H.stat == DEAD && is_station_level(H.z) && !is_type_in_typecache(A, medical_areas_morgue)) // If person is dead and corpse is NOT in one of these areas
-			return ..()
+			return FALSE
 	return TRUE
 
-/datum/objective/crew/chems //Ported from old Hippie
-	var/targetchem = "none"
-	var/datum/reagent/chempath
-	explanation_text = "Have (yell about this on GitHub, something broke) in your bloodstream when the shift ends."
-	jobs = "chiefmedicalofficer,chemist"
+/datum/objective/crew/chems
+	var/target_chemical
+	var/datum/reagent/target_chemical_name_obj
+	explanation_text = "Have at least (yell on GitHub if this breaks) units of X chemical in the smartfridge when the shift ends."
+	jobs = list(
+		JOB_NAME_CHIEFMEDICALOFFICER,
+		JOB_NAME_CHEMIST,
+	)
 
 /datum/objective/crew/chems/New()
 	. = ..()
-	chempath = get_random_reagent_id(CHEMICAL_GOAL_CHEMIST_BLOODSTREAM)
-	targetchem = chempath
+	target_chemical = get_random_reagent_id(CHEMICAL_GOAL_CHEMIST_USEFUL_MEDICINE)
+	// because initial() is picky
+	target_chemical_name_obj = target_chemical
+	target_amount = rand(10,40)
```

Comment:
this is very bad to cover most stuff. I suggest following this logic instead:
```
var/daum/reagent/R= target_chemical
target_amount = min(initial(R.metabolization_rate)*rand(20, 60), initial(R.overdose_threshold)-1)
```
20 ticks is 4u / 8u (0.2u / 0.4u rate)
60 ticks is 12u / 24u (0.2u / 0.4u rate)
It will not go over more than overdose.

Reply:
This is for MAKING the thing and storing it, not having in your bloodstream.

Reply:
or you can just put simple `min(target value, overdose)` rather than calculating metabolisation rate.

Reply:
> This is for MAKING the thing and storing it, not having in your bloodstream.
Chemists should be making more than one dose.

I don't think they'll make it as you intend. they'll make 40u of Oxandrolon then store it in the open fridge, and someone will just grab it to use without any doubt.

Reply:
Then they can keep it on their person if they really care about the objective. The point is not to get the objective, it is to encourage doing your job (making useful chems)

Reply:
no, this can be used as a justification for why they put 40u of oxandrolon at the public fridge, a kind of griefing justification.

Reply:
It's *total*, summed between all sources, not a 40u pill

Reply:
I know, but still I don't think giving any source that can be used as a justification isn't good. Well, It's up to you at this point.

Reply:
considering the old version encouraged you to make meth and other bad drugs, if a chemist is dumb and makes overdosing stuff they can suffer the IC consequences